massa-execution/src/interface_impl.rs

```diff
@@ -29,15 +38,73 @@ impl Interface for InterfaceImpl {
     fn get_module(&self, address: &String) -> Result<Vec<u8>> {
         let bytecode = {
             let address = Address::from_str(address)?;
-            let context = self
-                .context
-                .lock()
-                .expect("Failed to acquire lock on context.");
+            let context = context_guard!(self);
             context.ledger_step.get_module(&address)
         };
         match bytecode {
             Some(bytecode) => Ok(bytecode),
             _ => bail!("Error bytecode not found"),
         }
     }
+
+    /// Requires a new address that contains the sent bytecode.
+    ///
+    /// Generate a new address with a concatenation of the firsts N bytes of the blockId and the firsts N bytes of the block creator address.
+    /// Where N = ADDRESS_SIZE_BYTES / 2
+    ///
+    /// Insert in the ledger the given bytecode in the generated address
+    fn create_module(&self, module: &Bytecode) -> Result<assembly_simulator::Address> {
+        let (block_id, index) = {
+            let context = context_guard!(self);
+            (context.opt_block_id, context.call_stack.len())
+        };
+        let (block_id, index) = match (block_id, u8::try_from(index)) {
+            (Some(block_id), Ok(index)) => (block_id, index),
+            _ => bail!("Failed to read current context"),
+        };
+        let mut data = block_id.to_bytes().to_vec();
+        data.push(index);
```

Comment:
index is an u8 ? that may overflow. It should be u64 (or at least u32), use u64::to_be_bytes() for the byte conversion to remain deterministic

Reply:
modified